airflow/gcp/hooks/bigquery.py

```diff
@@ -1430,6 +1430,38 @@ def cancel_query(self) -> None:
                               self.running_job_id)
                 time.sleep(5)
 
+    def get_dataset_tables(self, dataset_id: str, project_id: Optional[str] = None,
+                           max_results: Optional[int] = None, page_token: Optional[str] = None) -> Dict:
```

Comment:
Can we make the return type more precise? For example `Dict[str, Dict].

Reply:
Changed it to `Dict[str, Union[str, int, List]]`.